src/application/Export.java

```diff
@@ -47,20 +156,47 @@ public String getFilesDirPath() {
         return "";
     }
 
+    private void delete_xml(String name){
+        File file = new File(targetPath+name);
+        file.delete();
+    }
+
+
+
     public boolean saveLessonsToFile(ArrayList<Lesson> lessons) {
+        xml = lessons;
+        try {
+            chooseTargetPath();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
         DataController dcToExport = CreateDCWithLessons(lessons);
 
         try {
+
             JAXBContext context = JAXBContext.newInstance(DataController.class);
             Marshaller m = context.createMarshaller();
             m.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, Boolean.TRUE);
-            m.marshal(dcToExport, new File(targetPath));
+            System.out.println(targetPath);
+            m.marshal(dcToExport, new File(targetPath+"/random.xml"));
+
+            try {
+                zipDirectory(targetPath, targetPath+"/data.zip","all");
```

Comment:
tejto celej metode trosku nerozumiem, ona ti sluzila na testy? mi to pripada ako keby si vytvoril zip do 'targetPath' ale o par riadkov neskor si to vymazal aj s nejakym 'random.xml'